flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -55,154 +53,186 @@ class SentryAssetBundle implements AssetBundle {
+    );
+    span?.setData('file.path', key);
+    // ignore: invalid_use_of_internal_member
+    span?.origin = SentryTraceOrigins.autoFileAssetBundle;
 
-      // This future is intentionally not awaited. Otherwise we deadlock with
-      // the completer.
-      // ignore: unawaited_futures
-      runZonedGuarded(() async {
-        final data = await _loadStructuredBinaryDataWrapper(
-          key,
-          (value) async => await _wrapBinaryParsing(parser, value, key, span),
-        );
-        span?.status = SpanStatus.ok();
-        completer.complete(data);
-      }, (exception, stackTrace) {
-        completer.completeError(exception, stackTrace);
-      });
+    if (updateInnerSpan != null) {
+      updateInnerSpan(span);
+    }
 
-      T data;
-      try {
-        data = await completer.future;
+    return _wrapWithCompleter(
+      action: loadFunction,
+      onSuccess: (data) {
         _setDataLength(data, span);
         span?.status = const SpanStatus.ok();
-      } catch (e) {
-        span?.throwable = e;
-        span?.status = const SpanStatus.internalError();
-        rethrow;
-      } finally {
-        await span?.finish();
-      }
-      return data;
-    });
+        span?.finish(); // Do NOT await
+      },
+      onError: (error, stackTrace) {
+        span?.throwable = error;
+        span?.status = SpanStatus.internalError();
+        span?.finish(); // Do NOT await, as this will lead to flickering.
+      },
+    );
   }
 
-  @override
-  Future<String> loadString(String key, {bool cache = true}) {
-    return Future<String>(() async {
-      final span = _hub.getSpan()?.startChild(
-            'file.read',
-            description: 'AssetBundle.loadString: ${_fileName(key)}',
-          );
-
-      span?.setData('file.path', key);
-      span?.setData('from-cache', cache);
-      // ignore: invalid_use_of_internal_member
-      span?.origin = SentryTraceOrigins.autoFileAssetBundle;
+  Future<T> _wrapParser<T>(
+    ISentrySpan? outerSpan,
+    String key,
+    Future<T> Function() parserFunction,
+  ) {
+    final span = outerSpan?.startChild(
+      'serialize.file.read',
+      description: 'parsing "$key" to "$T"',
+    );
+    // ignore: invalid_use_of_internal_member
+    span?.origin = SentryTraceOrigins.autoFileAssetBundle;
 
-      String? data;
-      try {
-        data = await _bundle.loadString(key, cache: cache);
-        span?.status = SpanStatus.ok();
-      } catch (exception) {
-        span?.throwable = exception;
+    return _wrapWithCompleter(
+      action: parserFunction,
+      onSuccess: (data) {
+        span?.status = const SpanStatus.ok();
+        span?.finish(); // Do NOT await
+      },
+      onError: (error, stackTrace) {
+        span?.throwable = error;
         span?.status = SpanStatus.internalError();
-        rethrow;
-      } finally {
-        await span?.finish();
+        span?.finish(); // Do NOT await, as this will lead to flickering.
+      },
+    );
+  }
```

Comment:
afaict it's fine to not await here

- we don't depend on the span to finish
- we don't influence the assets

so I'd say it's fine? but best to double check

Reply:
All tests do pass, assuming we cover all relevant cases, this should be fine.